### DISCUSSION TITLE
How to pass requested ACR values?

### DISCUSSION THREAD

**Discussion by mworrell** [erlef/oidcc#350] [Q&A] [answered]
Do you have an example how we can pass the requested ACR values to `oidcc:create_redirect_url/4` ?

**Comment by maennchen** [comment, marked as answer]
Currently the following request parameters are set automatically:

* `response_type`
* `response_mode`
* `scope`
* `redirect_uri`
* `client_id`
* `state`
* `nonce`
* `purpose`
* PKCE related
* PAR related
* DPoP related
* Request Object related

All other parameters like `prompt`, `acr_values` or custom provider parameters can be set using the `url_extension` option.

```erlang
{ok, Url} = oidcc_authorization:create_redirect_url(ClientContext, #{
  redirect_uri => RedirectUri,
  client_id => ClientId,
  url_extension => [{<<"acr_values">>, <<"acr:value">>}]
})
```

**Reply by mworrell** [reply]
Thanks!!!

**Reply by mworrell** [reply]
And the `"acr:value"` is just a (space separated) string of ACR values, isn't it?

**Reply by mworrell** [reply]
BTW `scopes` or `scope` ?? I am now using `scopes`

**Reply by maennchen** [reply]
@mworrell Correct

**Reply by maennchen** [reply]
> BTW `scopes` or `scope` ?? I am now using `scopes`

The parameter is called `scope` in the request. The option in the `create_redirect_url` opts is called `scopes` however since it's a list.
https://github.com/erlef/oidcc/blob/299114248949c8c83d5a853f1340991733c763c0/src/oidcc_authorization.erl#L22

https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest
